packages/@adobe/spectrum-css-temp/components/table/index.css

```diff
@@ -245,22 +248,75 @@ svg.spectrum-Table-sortedIcon {
     outline: 0;
   }
 
-  /* The focus ring must be on top of the first cell, so needs to be absolutely positioned. */
+  /* The focus ring should be sticky so it is visible at all horizontal scroll positions. */
   &:focus-ring {
     &:before {
+      display: inline-block;
       content: '';
-      position: absolute;
-      top: 0;
-      bottom: 0;
+      position: sticky;
       left: 0;
+      height: 100%;
       width: var(--spectrum-selectlist-border-size-key-focus);
-      z-index: 3;
+      z-index: 4;
+      margin-inline-end: calc(var(--spectrum-selectlist-border-size-key-focus) * -1)
+    }
+
+    .spectrum-Table-cellWrapper:first-of-type {
+      display: inline-block;
+      vertical-align: top;
+    }
+  }
+
+  &.spectrum-Table-row--highlightSelection {
+    /* Alternative to border-left/right. Using box shadow for the row left/right since they don't take room unlike border which would cause wiggles
+     * in the hightlight case and displace the sticky indicator.
+     */
+    &.is-selected {
+      /* this is the box shadow for left and right border outline */
+      &:after {
+        box-shadow: inset 1px 0 0 0 var(--spectrum-global-color-blue-500), inset -1px 0 0 0 var(--spectrum-global-color-blue-500);
+        content: '';
+        display: block;
+        position: absolute;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        top: 0;
+        z-index: 3;
+        pointer-events: none;
+      }
+    }
+    /* Add additional box shadow to top of the first row since border-top will push the bottom border out of view in Safari and Firefox. */
+    &.spectrum-Table-row--firstRow {
+      &.is-selected {
+        &:after {
+          border-start-start-radius: var(--spectrum-table-row-start-end-border-radius);
+          border-start-end-radius: var(--spectrum-table-row-start-end-border-radius);
+          box-shadow: inset 1px 0 0 0 var(--spectrum-global-color-blue-500), inset -1px 0 0 0 var(--spectrum-global-color-blue-500), inset 0 1px 0 0 var(--spectrum-global-color-blue-500);
```

Comment:
new strategy to fix the weird rendering that Safari and Firefox for the first/second row that @devongovett noticed. Used a box shadow to render the top blue border instead of using border top since border top pushed the bottom border out of view. What do ppl think? If this feels ok, maybe I could replace the bottom border with a box-shadow as well